### ISSUE TITLE
Running digilib on Raspberry Pi 3

### ISSUE THREAD

**Issue by kev1807** [robcast/digilib#16]
As promised yesterday, here are some further enquiries for running digilib on a raspberry pi 3: 
I am working with Putty.
OS: Raspbian Stretch, Kernel-Version 4.14 , image downloaded yesterday upgrade & update done. Pi says its up-to-date 

After I managed to get Tomcat up& running, I used followed the documentation (https://robcast.github.io/digilib/install-digilib.html) 
as follows
`cd /home/pi/Tomcat/apache-tomcat-9.0.12`
`wget https://github.com/robcast/digilib/releases/download/release-2.5.6/digilib-webapp-2.5.6-srv3.war`
`mv digilib-webapp-2.5.6-srv3.war digilib.war`

So, at this point my structure inside ~/Tomcat/webapps looks like this:
![grafik](https://user-images.githubusercontent.com/15955790/46719295-d90fe800-cc6d-11e8-8ca3-1864bc207030.png)

then I restarted tomcat using 
`sudo /etc/init.d/tomcat stop`
`sudo /etc/init.d/tomcat start`

and tried accessing it via browser, on my computer I am working on, not the Pi! resulting in a 404:
![grafik](https://user-images.githubusercontent.com/15955790/46718842-c3e68980-cc6c-11e8-9f37-b6f4f7f6b50c.png)

The apache-tomcat-9.0.12 is running properly, as 192..168.0.24:8080 (my Pis IP) redirects to the tomcat standard-page.

Is there a fast solution? 
I checked using the Pi directly: Also not working. 

Tomcat Installation:
([this](http://androidsrc.net/installing-tomcat8-raspberry-pi-3/) worked for me, however the autostart script seems not to work, although I think I adjusted every bit of it to match my needs.My Pi will likely be running 24/7 in the end, so that's not too much trouble (if any ideas, feel welcome to comment about it!)

**Comment by funkyfuture** [comment]
this is rather guessed as i don't have my pi at hand now: Debian uses `systemd` as service manager, these SysV ini scripts might be there for legacy reasons. anyway, the package should have installed, enabled and started that service. what does `sudo systemctl status tomcat` say? (oh, i just saw that you installed from an external source, why not from the Debian repos?)

a general hint regarding the limited cpu and memory resources: use `nginx` as proxy.

if you really want to run eXist-db on a Pi, restart it regularly. at least the versions in the `3.x` branch steadily increase their used memory.

**Comment by kev1807** [comment]
I am currently not at my Computer, so I cannot test this. I'll reply in the next few days, since I'm busy with my day-job for the next two days or so ... 
why not from Debian repos: probably because I figured the official source was up-to-date, other words: I was blindly following the tutorial, as I simply wanted it running.

I aim for exist-db on my Windows-Machine and making the Pi an IIF-Image Server in the future, for testing my ideas, so no exist-db on the pi. The Pi will only carry images, the rest will remain on another computer, thus trying to resemble an actual IIIF-conform setup, with the images being streamed to the website. (I'll someday hopefully be able to show something somewhere!)


As for the main issue of digilib not starting: It was a Tomcat-problem, as Tomcat seems to only allow localhost:8080 after the installation, so I had to adapt I believe the webapps/manager/META-INF/config.xml <Valve ...> (from memory, so no guarantee that its correct!) 

It's up to someone else to state whether this issue should be closed, since the main issue is solved and a rather remote question, not directly connected to digilib is not, so I'll leave it open, but it may be closed, if decided so! 

Thanks for your help anyway :-)
